### PR TITLE
[language] increase deserializer test coverage

### PR DESCRIPTION
**Pull request by dariorussi**
Covering few malformed binaries in deserializer.
This increases the coverage a bit (we are over 92%).
There are few more error case I want to think about, as it
gets harder and harder to come up with test cases

### CODE REVIEW

File: language/vm/src/unit_tests/signature_token_tests.rs [hunk 1]
```diff
@@ -14,29 +14,32 @@ fn serialize_and_deserialize_nested_types_max() {
...
+        let mut cursor = Cursor::new(binary.as_inner());
+        load_signature_token(&mut cursor).expect("deserialization should succeed");
+    }
 }
 
 #[test]
 fn serialize_nested_types_too_deep() {
     let mut ty = SignatureToken::Struct(StructHandleIndex::new(0));
-    for _ in 0..SIGNATURE_TOKEN_DEPTH_MAX {
+    for _ in 1..SIGNATURE_TOKEN_DEPTH_MAX {
```
Comment (tzakian) [inline language/vm/src/unit_tests/signature_token_tests.rs:28]: Out of curiosity, is there also another test that makes sure that nested types are allowed up to this limit?

**Review by tzakian — APPROVED** [review]
Looks good to me

File: language/vm/src/unit_tests/signature_token_tests.rs (hunk 1 above)
Comment (dariorussi) [reply language/vm/src/unit_tests/signature_token_tests.rs:28]: no and I plan to add one, absolutely good point!
But overall I realized that coverage in deserializer is pretty high so I figured I'd spend the next few days looking for other places where we may not have a good enough coverage

**Pull request closed without merging**